src/frontend/src/eth/components/send/SendAmount.svelte

```diff
@@ -1,91 +1,95 @@
 	export let nativeEthereumToken: Token;
 
-	let insufficientFundsError: string | undefined;
+	let insufficientFundsError: InsufficientFundsError | undefined = undefined;
 
 	$: insufficientFunds = nonNullish(insufficientFundsError);
 
 	const { feeStore: storeFeeData } = getContext<FeeContext>(FEE_CONTEXT_KEY);
 	const { sendTokenDecimals, sendBalance, sendTokenId } = getContext<SendContext>(SEND_CONTEXT_KEY);
 
-	const validate = () => {
-		if (invalidAmount(amount)) {
-			insufficientFundsError = undefined;
+	let fee: BigNumber | undefined | null = undefined;
+
+	let timer: NodeJS.Timeout | undefined;
+
+	$: $storeFeeData,
+		(() => {
+			timer = setTimeout(() => {
+				fee = getFee($storeFeeData);
+			}, 500);
+		})();
+
+	onDestroy(() => {
+		if (isNullish(timer)) {
 			return;
 		}
 
+		clearInterval(timer);
+	});
+
+	$: customValidate = (userAmount: BigNumber) => {
 		if (isNullish($storeFeeData)) {
 			insufficientFundsError = undefined;
 			return;
 		}
 
-		const userAmount = parseToken({
-			value: `${amount}`,
-			unitName: $sendTokenDecimals
-		});
-
 		// If ETH, the balance should cover the user entered amount plus the min gas fee
 		if (isSupportedEthTokenId($sendTokenId)) {
 			const total = userAmount.add(minGasFee($storeFeeData));
 
 			if (total.gt($sendBalance ?? BigNumber.from(0n))) {
-				insufficientFundsError = $i18n.send.assertion.insufficient_funds_for_gas;
-				return;
+				insufficientFundsError = new InsufficientFundsError(
+					$i18n.send.assertion.insufficient_funds_for_gas
+				);
+				return insufficientFundsError;
 			}
 
 			insufficientFundsError = undefined;
-			return;
+			return insufficientFundsError;
 		}
 
 		// If ERC20, the balance of the token - e.g. 20 DAI - should cover the amount entered by the user
 		if (userAmount.gt($sendBalance ?? BigNumber.from(0n))) {
-			insufficientFundsError = $i18n.send.assertion.insufficient_funds_for_amount;
-			return;
+			insufficientFundsError = new InsufficientFundsError(
+				$i18n.send.assertion.insufficient_funds_for_amount
+			);
+			return insufficientFundsError;
 		}
+
 		// Finally, if ERC20, the ETH balance should be less or greater than the max gas fee
 		const maxFee = maxGasFee($storeFeeData);
 		const ethBalance = $balancesStore?.[nativeEthereumToken.id]?.data ?? BigNumber.from(0n);
-
 		if (nonNullish(maxFee) && ethBalance.lt(maxFee)) {
-			insufficientFundsError = $i18n.send.assertion.insufficient_ethereum_funds_to_cover_the_fees;
-			return;
+			insufficientFundsError = new InsufficientFundsError(
+				$i18n.send.assertion.insufficient_ethereum_funds_to_cover_the_fees
+			);
+			return insufficientFundsError;
 		}
-
-		insufficientFundsError = undefined;
 	};
 
-	const debounceValidate = debounce(validate);
-
-	$: amount, $storeFeeData, debounceValidate();
+	let calculateMax = (): number | undefined => {
+		return getMaxTransactionAmount({
+			balance: $sendBalance?.toBigInt(),
+			fee: fee?.toBigInt(),
+			token: nativeEthereumToken
```

Comment:
I have the feeling that's not the correct token. This is always Ethereum so, it probably always uses 18 decimals. I guess we need to use the `sendToken`?

Reply:
Indeed, we could generalize and use the `token.derived` here too. I double-checked and it returns the current `routeToken` that should point to the correct token.

One thing to discuss is to correct on the parent components too, since it was treated as prop since the component SendTokenWizard.

Reply:
Actually, I think it must be `sendToken` here and not the route token. Given that we have those cross-conversion between token, `sendToken` ensures that it's the correct "source".

Reply:
And `nativeEthereumToken` is definitely incorrect.

Reply:
corrected